typed-racket-lib/typed-racket/typecheck/toplevel-trampoline.rkt

```diff
@@ -66,22 +66,23 @@
           (local-expand/capture* #'e 'top-level (kernel-form-identifier-list))))
        (syntax-parse head-expanded
          #:literal-sets (kernel-literals)
-         [(begin (define-values (n) _) ...
-                 (~and (~or _:ignore^ _:ignore-some^)
-                       (~not (~or _:tr:class^
-                                  _:tr:unit^
-                                  _:tr:unit:invoke^
-                                  _:tr:unit:compound^
-                                  _:tr:unit:from-context^))))
-          head-expanded]
```

Comment:
It handled all ignored forms including those that start with `begin` and just returned them, but it turns out this is incorrect since all forms (including ignored forms) need to be local expanded so that any lifts that they do are resolved.

So instead this delegates to the other two clauses, which will either open up begins but propagate syntax properties so that ignored things are ignored properly. Alternatively if it's not a `begin` form then the actual typechecker (and optimizer) kick in after local expansion, but since the ignore property is there it should work properly.

There is a caveat that this will transfer all syntax properties, so TR shouldn't put any properties on top-level `begin`s that are invalid when transferred to subforms.